test/numbers.jl

```diff
@@ -2914,7 +2914,7 @@ Base.literal_pow(::typeof(^), ::PR20530, ::Val{p}) where {p} = 2
     p = 2
     @test x^p == 1
     @test x^2 == 2
-    @test [x,x,x].^2 == [2,2,2]
+    @test_broken [x, x, x].^2 == [2, 2, 2] # literal_pow violates referential transparency
```

Comment:
What result does this give now?

Reply:
On master, I get `[x,x,x].^2 == [2,2,2]` but I guess this PR breaks this? Temporarily or permanently?

Reply:
I don't think `literal_pow` really played well with `broadcast` and dot syntax to begin with. It was a little unpredictable when `literal_pow` or regular `^` would be called...

Reply:
Yes, on this PR you get `[1, 1, 1]`.